src/main/kotlin/org/rust/cargo/project/module/persistence/CargoProjectDataService.kt

```diff
@@ -11,23 +11,23 @@ import com.intellij.openapi.project.Project
 import org.rust.cargo.CargoConstants
 import org.rust.cargo.CargoProjectDescription
 import org.rust.cargo.util.getService
+import java.io.Serializable
 
 /**
- * Populates [CargoModuleTargetsService] on project import
+ * Populates [CargoModuleService] on project import
  */
-class CargoProjectDataService : AbstractProjectDataService<CargoProjectDescription.Target, Module>() {
-    override fun getTargetDataKey(): Key<CargoProjectDescription.Target> = CargoConstants.KEYS.TARGET
+class CargoProjectDataService : AbstractProjectDataService<CargoModuleData, Module>() {
+    override fun getTargetDataKey(): Key<CargoModuleData> = CargoConstants.KEYS.CARGO_MODULE_DATA
 
-    override fun postProcess(toImport: Collection<DataNode<CargoProjectDescription.Target>>,
+    override fun postProcess(toImport: MutableCollection<DataNode<CargoModuleData>>,
```

Comment:
Because Kotlin generates `MutableCollection` if you override a method whose argument is Java's `Collection`, and you have to change it to immutable version manually. Fixed